src/Checkbox/utils.ts

```diff
@@ -0,0 +1,68 @@
+import type { Theme } from '../shared/theme.types';
+
+export const getBackgroundColorFromState = ({
+  checked,
+  disabled,
+  theme,
+  $indeterminate,
+}: {
+  checked?: boolean;
+  disabled?: boolean;
+  theme: Theme;
+  $indeterminate?: boolean;
+}) => {
+  if (disabled && $indeterminate) {
+    return theme.palette.purple[400];
+  }
+  if (!disabled && $indeterminate) {
+    return theme.palette.purple.main;
+  }
+  if (checked && !disabled) {
+    return theme.palette.purple.main;
+  }
+  if (disabled) {
+    return theme.palette.neutral[300];
+  }
+  return 'white';
```

Comment:
Maybe we should add white and black to the theme? Something like:
```suggestion
  return theme.palette.common.white;
```

Reply:
leave it like that for now?